server/server.go

```diff
@@ -192,6 +200,13 @@ func (s *Server) WithCompilerErrorLimit(limit int) *Server {
 	return s
 }
 
+// WithDiagnosticsBufferSize sets the size on the diagnostics buffer stored by the
+// server.
+func (s *Server) WithDiagnosticsBufferSize(size int) *Server {
```

Comment:
I had imagined that the buffer size was essentially an exported server property (so that we wouldn't run out of memory). If the buffer is only used in the server, an interface seems unneeded.